macos/FluentUI/Button.swift

```diff
@@ -0,0 +1,230 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import AppKit
+
+/// Indicates what style our button is drawn as
+@objc public enum ButtonStyle: Int, CaseIterable {
+    case primaryFilled	// Solid fill color
+    case primaryOutline	// Clear fill color, solid outline
+    case borderless		// Clear fill color, clear outline
+}
+
+// MARK: - Button
+
+/// A fluent styled button, with hover effects and a corner radius.
+@objc(MSFButton)
+open class Button: NSButton {
+
+	@objc public init(title: String, style: ButtonStyle = .primaryFilled) {
+		super.init(frame: .zero)
+		self.title = title
+		self.style = style
+		initialize()
+    }
+	
+	@objc public init(image: NSImage, style: ButtonStyle = .primaryFilled) {
+		super.init(frame: .zero)
+		self.style = style
+		initialize()
+		// Image must be set after `wantsLayer = true` so that the image is on the layer and not the underlying NSButtonCell
+		imagePosition = .imageOnly
+		self.image = image
+    }    
+	
+	public override init(frame frameRect: NSRect) {
+		super.init(frame: frameRect)
+		initialize()
+	}
+
+	@available(*, unavailable)
+	required public init?(coder decoder: NSCoder) {
+		preconditionFailure()
+	}
+	
+	private func initialize() {
+		// Do common initialization work
+		isBordered = false
+		wantsLayer = true
+		update()
+	}
+	
+	override public var image: NSImage? {
+		didSet {
+			if imagePosition != .imageOnly {
+				preconditionFailure("We currently do not support adding both an image and a title. Please set the imagePosition to .imageOnly before setting an image")
+			}
```

Comment:
Probably more of a TODO than a preconditionFailure? This definitely seems like something we will support so at the very least we should remove the instructions for how to workaround it and replace with anything useful that could help someone contribute a fix if they run into this

Reply:
@markavitale mentioned we could reference a github issue here

Reply:
Removed the need of this `preconditionFailure` by making sure the imagePosition is imageOnly in willSet. Agree this is something we definitely want to add later on